### PR TITLE
dev-cmd/bump: add `--no-fork` switch

### PR DESCRIPTION
**Pull request by p-linnane**
Signed-off-by: Patrick Linnane <patrick@linnane.io>- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
This PR brings the `--no-fork` switch in `bump-cask-pr` and `bump-formula-pr` to `brew bump`. Given that maintainers are now using branches within the main repo, this will make things easier in two ways:

1. Maintainers can make manual changes to formulae or casks, save them, and then run `brew bump --open-pr --no-fork foo` to handle the rest of the PR process with their local changes incorporated.
2. We can use this for BrewTestBot's autobump PR's. This will reduce the thousands of stale branches in the BrewTestBot forks (which I still plan to deal with).

Example PR opened using this branch: https://github.com/Homebrew/homebrew-core/pull/163948

As usual, thanks to @Bo98 for his guidance so I could put this together.

### CODE REVIEW

**Review by Bo98 — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @p-linnane! Wasn't worth blocking this PR but I think ideally this would intelligently figure out whether to use a fork or not based on maintainer permissions. I'll maybe look at that in future 👍🏻

**Comment by p-linnane** [comment]
That sounds awesome. It's a bit out of my ability I think, so looking forward to seeing if you take a look!

**Pull request merged**
